api_loader.go

```diff
@@ -551,10 +551,29 @@ func loadApps(specs []*APISpec, muxer *mux.Router) {
 
 	chainChannel := make(chan *ChainObject)
 
+	// Remove inactive APIs from the specs
+	i := 0 // output index
+	for _, spec := range specs {
+		if !spec.Active {
+			log.WithFields(logrus.Fields{
+				"prefix":   "main",
+				"api_name": spec.Name,
+				"domain":   spec.Domain,
+			}).Info("Skipping Inactive.")
+
+			continue
+		}
+		// copy & increment index
+		specs[i] = spec
+		i++
+	}
+	specs = specs[:i]
+
```

Comment:
You can simplify this by just doing this logic in the `for i, spec := range specs` loop. All you would need is:

```
		if !spec.Active {
 			log.WithFields(logrus.Fields{
 				"prefix":   "main",
 				"api_name": spec.Name,
 				"domain":   spec.Domain,
 			}).Info("Skipping Inactive.")
                        loadList[i] = &ChainObject{Skip: true}
 			continue
 		}
```